src/averageAges.js

```diff
@@ -14,12 +14,29 @@
  *
  * @return {number}
  */
+
+function filterSex(array, sex) {
+  return array.filter(person => person.sex === sex);
+};
+
+function calcAvarage(number, divider) {
+  return Number((number / divider).toFixed(2));
+}
+
 function calculateMenAverageAge(people, century) {
-  // write code here
-  // learn how to use array methods like .filter .map .some .every .find .reduce
-  // avoid using loop and forEach
-  // replace `if ()` statement with &&, || or ?:
-  // without nesting
+  let mens = filterSex(people, 'm');
+
+  if (century) {
+    mens = mens.filter(
+      person => Math.ceil(person.died / 100) === century
+    );
+  }
+
+  const totalAge = mens.reduce(
+    (sum, person) => sum + (person.died - person.born), 0
+  );
+
+  return calcAvarage(totalAge, mens.length);
```

Comment:
and here, instead of summing, get an array of differences and send into calcAverage(ages differences )